portfolio/src/main/java/com/google/sps/servlets/LeaderboardServlet.java

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+
+/** Servlet responsible for handling comments **/
+@WebServlet("/leaderboards")
+public class LeaderboardServlet extends HttpServlet {
+    private DatastoreService datastore;
+    private LinkedHashMap<String, Long> winStreaks;
+    private HashMap<String, String> userNickNames;
+
+    @Override
+    public void init() {
+        datastore = DatastoreServiceFactory.getDatastoreService();
+        winStreaks = new LinkedHashMap<>();
+        userNickNames = new HashMap<>();;
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        userNickNames.clear();
+        winStreaks.clear();
+        Query query = new Query("PingPongEntity").addSort("longestWinStreak", Query.SortDirection.DESCENDING);
```

Comment:
Maybe another option would be to do a joined query, instead of querying 2 separate times.